packages/venia-concept/src/components/ErrorView/errorView.js

```diff
@@ -5,15 +5,21 @@ import { loadingIndicator } from 'src/components/LoadingIndicator';
 const messages = new Map()
     .set('loading', loadingIndicator)
     .set('notFound', 'That page could not be found. Please try again.')
-    .set('internalError', 'Something went wrong. Please try again.');
+    .set('internalError', 'Something went wrong. Please try again.')
+    .set(
+        'outOfStock',
+        'This Product is currently out of stock. Please try again later.'
+    );
 
 class ErrorView extends Component {
     render() {
-        const { loading, notFound } = this.props;
+        const { loading, notFound, outOfStock } = this.props;
         const message = loading
             ? messages.get('loading')
             : notFound
             ? messages.get('notFound')
+            : outOfStock
+            ? messages.get('outOfStock')
```

Comment:
We may want to allow an error string rather than using this prop. This will allow us to separate the error messages and types from the `ErrorView` implementation.

```js
export const ERROR_TYPES = {
  'LOADING': 'loading',
  'OUT_OF_STOCK': 'outOfStock',
  ...
};
export const errorMap = new Map({
  'loading': loadingIndicator,
  'outOfStock': 'This Product is currently out of stock. Please try again later.',
  'default': 'Something went wrong. Please try again.'
  ...
});

class ErrorView extends Component {
  render() {
    const { type } = this.props;
    const key = ERROR_TYPES[type] || 'default';
    return <h1>{errorMap[key]}</h1>;
  }
} 
```

Reply:
Yeah the `ErrorView` certainly could use a refactor. I already felt like adding `outOfStock` was a little scope creep here and refactoring `ErrorView` entirely would be out of scope entirely.

Reply:
I can create a follow up ticket to refactor `ErrorView` for sure

Reply:
That's cool -- though the component is super small and I feel like the effort to refactor now is less than the effort to create, groom and work on it later. But w/e you want to do is cool, you're the one doing this work now :D

Reply:
A fair point, lol.

I created #1308 just to keep things moving though.